test_change_reclaim_policy_of_pv failing

Test test_change_reclaim_policy_of_pv is failing because multi_pvc_factory_fixture is not using the  parameter storageclass which is passed from the test. So the storage class and PV will be always created with reclaim policy Delete. 
AssertionError: Reclaim policy of pvc-086b3f68-9984-4202-80f6-f4400f9c92fe is Delete instead of Retain

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1103,10 +1103,9 @@ def multi_pvc_factory_fixture(
             status_tmp = ""
 
         project = project or project_factory()
-        if interface == constants.CEPHBLOCKPOOL:
-            storageclass = helpers.default_storage_class(interface_type=interface)
-        elif interface == constants.CEPHFILESYSTEM:
-            storageclass = helpers.default_storage_class(interface_type=interface)
+        storageclass = storageclass or helpers.default_storage_class(
+            interface_type=interface
+        )
 
         access_modes = access_modes or [constants.ACCESS_MODE_RWO]
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -448,10 +448,14 @@ def pvc_factory_fixture(
             project = project or active_project or project_factory()
             active_project = project
             if interface == constants.CEPHBLOCKPOOL:
-                storageclass = helpers.default_storage_class(interface_type=interface)
+                storageclass = storageclass or helpers.default_storage_class(
+                    interface_type=interface
+                )
                 active_rbd_storageclass = storageclass
             elif interface == constants.CEPHFILESYSTEM:
-                storageclass = helpers.default_storage_class(interface_type=interface)
+                storageclass = storageclass or helpers.default_storage_class(
+                    interface_type=interface
+                )
                 active_cephfs_storageclass = storageclass
 
             pvc_size = f"{size}Gi" if size else None

